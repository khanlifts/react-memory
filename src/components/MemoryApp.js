import React from 'react';
import Card from './Card';
import AddContent from './AddContent';

export default class MemoryApp extends React.Component {
  state = {
    content: []
  };
  handleAddContent = (content) => {
    // if there is an empty string
    if (!content) {
      return 'Valid value please';
    // if there is already the same entry. indexOf returns index of option in array
    // - 1 indicates a new entry. all other entries already have an index
  } else if (this.state.content.indexOf(content) > -1) {
      return 'This option already exists.';
    }
    // use concat return new array with merged content of original arrays
    this.setState((prevState) => ({ content: prevState.content.concat(content) }));
  };
  render() {
    return (
      <div className="app__container">
        <h2>This renders correctly</h2>
        <AddContent
          options={this.state.content}
          handleAddContent={this.handleAddContent}
          />
        <Card />
      </div>
    );
  }
}
