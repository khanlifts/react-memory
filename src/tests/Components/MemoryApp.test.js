import React from 'react';
import { shallow } from 'enzyme';
import MemoryApp from '../../components/MemoryApp';

test('should render component correctly', () => {
  const wrapper = shallow(<MemoryApp />);
  expect(wrapper).toMatchSnapshot();
});

test('should render gitHubCorner', () => {
  const wrapper = shallow(<MemoryApp />);
  // expect(wrapper.find('.app__container')).to.have.length(1);
});
