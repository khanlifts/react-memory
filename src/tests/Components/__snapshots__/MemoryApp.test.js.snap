// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render component correctly 1`] = `
<div
  className="app__container"
>
  <h1>
    Reactive Memory
  </h1>
  <GithubCorner
    ariaLabel="Open GitHub project"
    bannerColor="#151513"
    className="gitHubCorner"
    direction="right"
    href="https://github.com/khanlifts/react-memory"
    octoColor="#fff"
    size="100"
  />
  <Action
    content={Array []}
    handleAddContent={[Function]}
    handleDeleteAllContent={[Function]}
    handleDuplicateContent={[Function]}
    handleShuffleContent={[Function]}
  />
  <Grid
    content={Array []}
    handleDeleteContent={[Function]}
    handleSameContent={[Function]}
  />
</div>
`;
